src/Compilers/Core/Portable/Operations/InstanceReferenceKind.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Kind of reference for an <see cref="IInstanceReferenceOperation"/>.
+    /// </summary>
+    public enum InstanceReferenceKind
+    {
+        /// <summary>
+        /// Reference to a member defined on the type of this instance reference, potentially overridden by the implementation.
+        /// Used for C# <code>this</code> and VB <code>Me</code>.
+        /// </summary>
+        ThisOverridable,
+        /// <summary>
+        /// Reference to a member defined on the type of this instance reference, as if the member was defined with <code>NotOverridable</code>.
+        /// Used for VB <code>MyClass</code>.
+        /// </summary>
+        ThisNonOverridable,
+        /// <summary>
+        /// Reference to a member defined on the base type of this instance reference.
+        /// Used for C# <code>base</code> and VB <code>MyBase</code>.
+        /// </summary>
+        Base,
```

Comment:
>Base [](start = 8, length = 4)

I don't think we want to have this kind of distinction for the first tree items here. We have IsVirtual property to specify if invocation or a method reference is using virtual semantics. I suggest merging all three under RegularReceiver or something similar

Reply:
Actually, it feels like ContainingTypeInstance is probably going to reflect semantics more accurately.

---
In reply to: [175573219](https://github.com/dotnet/roslyn/pull/25322#discussion_r175573219) [](ancestors = 175573219)